File: src/dataset/python/queen-attack_exercism-ThomasZumsteg/claude/ClaudeSonnet4_queen_attack_v3.py
"""Queen chess moves"""

def validate(func):
    """Ensures valid chess positions"""
    def chess_checker(black, white):
        """Ensures valid chess positions"""
        if (black[0] | black[1] | white[0] | white[1]) & ~7 or black == white:
            raise ValueError
        return func(black, white)
    return chess_checker

@validate
def board(white, black):
    """Prints a board with the queens locations"""
    result = ['_' * 8] * 8
    result = list(result)
    
    w_row, w_col = white
    b_row, b_col = black
    
    if w_row == b_row:
        row = list(result[w_row])
        row[w_col] = 'W'
        row[b_col] = 'B'
        result[w_row] = ''.join(row)
    else:
        w_row_chars = list(result[w_row])
        w_row_chars[w_col] = 'W'
        result[w_row] = ''.join(w_row_chars)
        
        b_row_chars = list(result[b_row])
        b_row_chars[b_col] = 'B'
        result[b_row] = ''.join(b_row_chars)
    
    return result

@validate
def can_attack(black, white):
    """Checks if the two queens can attack eachother"""
    row_diff = abs(black[0] - white[0])
    col_diff = abs(black[1] - white[1])
    return row_diff == col_diff or row_diff == 0 or col_diff == 0

File: src/dataset/javascript/queen-attack_exercism-ThomasZumsteg/claude/ClaudeSonnet4_queen-attack_v3.js
var Queens = function(args) {
	this.white = (args && args.white) || [0, 3];
	this.black = (args && args.black) || [7, 3];
	if(this.white[0] === this.black[0] && this.white[1] === this.black[1])
		throw "Queens cannot share the same space";
}

Queens.prototype.toString = function() {
	var result = "";
	for(var i = 0; i < 8; i++) {
		for(var j = 0; j < 8; j++) {
			if(this.white[0] === i && this.white[1] === j) {
				result += "W";
			} else if(this.black[0] === i && this.black[1] === j) {
				result += "B";
			} else {
				result += "_";
			}
			if(j < 7) result += " ";
		}
		result += "\n";
	}
	return result;
};

Queens.prototype.canAttack = function() {
	var rowDiff = Math.abs(this.black[0] - this.white[0]);
	var colDiff = Math.abs(this.black[1] - this.white[1]);
	return rowDiff === 0 || colDiff === 0 || rowDiff === colDiff;
};

module.exports = Queens;